src/precice/SolverInterface.hpp

```diff
@@ -851,6 +851,11 @@ class SolverInterface {
    *             corresponding data values (dim * \p size)
    *
    * @pre IDs and coordinates need to have the correct size, which can be queried by @p getMeshVertexSize()
+   *
+   * @note This function can be called on received meshes as well as provided
+   * meshes. However, you need to call this function after @p initialize(),
+   * if the meshID corresponds to a received mesh, since the relevant mesh data
+   * is exchanged during the @p initialize() call.
```

Comment:
You can simply state this as a precondition

```
 * @pre initialize() has been called
```

Reply:
Yes, I guess it needs some more explanation. I like the more verbose variant. I changed the note to an `@pre` though.